impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultSettingsBuilder.java

```diff
@@ -203,6 +204,12 @@ private Settings readSettings(
         settings = interpolate(settings, request, problems);
```

Comment:
```suggestion
        settings = interpolate(decrypt(settingsSource, settings, request, problems), request, problems);
```
good example of this smell and why its avoidable causing high coupling.

- https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#avoidreassigningparameters
- https://www.geeksforgeeks.org/software-engineering-coupling-and-cohesion/

Reply:
thanks for it - it is existing code, so I want not introduce changes not required by this PR.
It can be done as separate PR.

Reply:
yes of course. just wanted to share and ask.